File: Programmers/스택&큐/프린터.js
const solution = (priorities, location) => {
  let p = priorities.map((v, i) => {
    return { index: i, value: v };
  });
  let answer = [];
  while (p.length > 0) {
    let first = p.shift();
    let others = p.filter((v) => {
      return v.value > first.value;
    });
    if (others.length > 0) p.push(first);
    else {
      answer.push(first);
      if (answer[answer.length - 1].index === location) {
        return answer.length;
      }
    }
  }
};

console.log(solution([2, 1, 3, 2], 2));

File: Programmers/Level2/프린터.py
from collections import deque

def solution(priorities, location):
    array = deque()
    for i in range(len(priorities)):
        array.append((i, priorities[i]))
    # index를 담을 배열
    result = []
    # 탐색 시작
    while array:
        # 가장 높은 priority
        maxPriority = max(array, key=lambda x: x[1])[1]
        # 맨 앞 순서
        currentIdx, currentPriority = array.popleft()
        # priority 비교 후 같으면 출력
        if currentPriority == maxPriority:
            result.append(currentIdx)
        # 작으면 맨 뒷 순서로
        else:
            array.append((currentIdx, currentPriority))

    return result.index(location) + 1

print(solution([2, 1, 3, 2], 2))